activerecord/lib/active_record/tasks/database_tasks.rb

```diff
@@ -96,11 +97,11 @@ def fixtures_path
       end
 
       def root
-        @root ||= Rails.root
+        @root ||= Rails.root if defined?(Rails.root)
```

Comment:
This and the `env` method are part of the easter goodies 🐰 🥚 goodies in this PR imo. 

The api for this class allows clients to do something like: 

``` ruby
root = ActiveRecord::Tasks::DatabaseTasks.root
env = ActiveRecord::Tasks::DatabaseTasks.env
puts "Hello from #{root} in #{env}"
```

However this would fail if the `Rails.root` or `Rails.env` are not defined.